### PR TITLE
coerce property values to correct type

### PR DESCRIPTION
**Pull request by rodesai**
This patch changes Command to coerce overwritten properties to the right
types. Previously, if a user provided a property that deserializes to a
different type by default (e.g. a property that is declared as Short will
deserialize to Integer by default), the command would fail to execute due
to a type mismatch.

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java [hunk 1]
```diff
@@ -117,7 +118,7 @@ public String getStatement() {
       justification = "overwriteProperties is unmodifiableMap()"
   )
   public Map<String, Object> getOverwriteProperties() {
-    return overwriteProperties;
+    return PropertiesUtil.coerceTypes(overwriteProperties, true);
```
Comment (guozhangwang) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java:121]: Just to clarify this is the actual and only fix of this PR right? All others seems to be cleanups with no logical changes.

File: ksqldb-common/src/main/java/io/confluent/ksql/properties/PropertiesUtil.java [hunk 2]
```diff
@@ -159,4 +163,47 @@ private static void throwOnBlackListedProperties(final Map<String, ?> properties
...
+    try {
+      final String stringValue = value == null
+          ? null
+          : value instanceof List
+              ? listToString((List<?>) value)
+              : String.valueOf(value);
+
+      return PROPERTY_PARSER.parse(key, stringValue);
+    } catch (final Exception e) {
+      throw new KsqlException("Failed to set '" + key + "' to '" + value + "'", e);
```
Comment (guozhangwang) [inline ksqldb-common/src/main/java/io/confluent/ksql/properties/PropertiesUtil.java:200]: This exception message is a bit unclear to me, should that be "failed to coerce type of value {} for key {}"?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java (hunk 1 above)
Comment (guozhangwang) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java:121]: Also why we would like to ignore PropertyNotFoundException in command, while not in the ksql request below?

Comment (rodesai) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java:121]: The statement should already have been validated at this point, and we've committed to executing it. We should just deserialize and pass it along to the engine.

File: ksqldb-common/src/main/java/io/confluent/ksql/properties/PropertiesUtil.java (hunk 2 above)
Comment (rodesai) [reply ksqldb-common/src/main/java/io/confluent/ksql/properties/PropertiesUtil.java:200]: Makes sense - it was just copypasta

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java (hunk 1 above)
Comment (guozhangwang) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java:121]: Ah you're right!

**Review by guozhangwang — APPROVED** [review]

**Comment by guozhangwang** [comment]
Took another look at the latest commit, still LGTM.

**Comment by rodesai** [comment]
I'm merging this. The build passed here: https://jenkins.confluent.io/job/confluentinc-pr/job/ksql/view/change-requests/job/PR-8765/7/

**Pull request merged**